core/trino-main/src/test/java/io/trino/operator/TestSimplePagesHashStrategy.java

```diff
@@ -0,0 +1,147 @@
+
+        SimplePagesHashStrategy strategy = createSimplePagesHashStrategy(mapType, ImmutableList.of(block));
+        Page page = new Page(block);
+
+        // This works because MapType is comparable.
+        assertEquals(strategy.hashRow(0, page), 451258269207618863L);
+    }
+
+    @Test
+    public void testRowEqualsRowWithIntegerType()
+    {
+        SimplePagesHashStrategy strategy = createSimplePagesHashStrategy(INTEGER, ImmutableList.of());
+
+        Page leftPage = new Page(new IntArrayBlock(1, Optional.empty(), new int[]{1234}));
+        Page rightPage1 = new Page(new IntArrayBlock(1, Optional.empty(), new int[]{1234}));
+        Page rightPage2 = new Page(new IntArrayBlock(1, Optional.empty(), new int[]{5678}));
+
+        // This works because IntegerType is comparable.
+        assertTrue(strategy.rowEqualsRow(0, leftPage, 0, rightPage1));
+        assertFalse(strategy.rowEqualsRow(0, leftPage, 0, rightPage2));
+    }
+
+    @Test
+    public void testRowEqualsRowWithMapType()
+    {
+        MapType mapType = new MapType(INTEGER, INTEGER, new TypeOperators());
+        SimplePagesHashStrategy strategy = createSimplePagesHashStrategy(mapType, ImmutableList.of());
+
+        Page leftPage = new Page(mapType.createBlockFromKeyValue(
+                Optional.empty(),
+                new int[]{0, 1},
+                new IntArrayBlock(1, Optional.empty(), new int[]{1234}),
+                new IntArrayBlock(1, Optional.empty(), new int[]{5678})));
+
+        Page rightPage1 = new Page(mapType.createBlockFromKeyValue(
+                Optional.empty(),
+                new int[]{0, 1},
+                new IntArrayBlock(1, Optional.empty(), new int[]{1234}),
+                new IntArrayBlock(1, Optional.empty(), new int[]{5678})));
+
+        Page rightPage2 = new Page(mapType.createBlockFromKeyValue(
+                Optional.empty(),
+                new int[]{0, 1},
+                new IntArrayBlock(1, Optional.empty(), new int[]{1234}),
+                new IntArrayBlock(1, Optional.empty(), new int[]{1234})));
+
+        // This works because MapType is comparable.
+        assertTrue(strategy.rowEqualsRow(0, leftPage, 0, rightPage1));
+        assertFalse(strategy.rowEqualsRow(0, leftPage, 0, rightPage2));
+    }
+
+    @Test
+    public void testCompareSortChannelPositionsWithIntegerType()
+    {
+        Block block = new IntArrayBlock(3, Optional.empty(), new int[]{1234, 5678, 1234});
+        SimplePagesHashStrategy strategy = createSimplePagesHashStrategy(INTEGER, ImmutableList.of(block));
+
+        // This works because IntegerType is orderable.
+        assertEquals(strategy.compareSortChannelPositions(0, 0, 0, 1), -1);
+        assertEquals(strategy.compareSortChannelPositions(0, 1, 0, 0), 1);
+        assertEquals(strategy.compareSortChannelPositions(0, 0, 0, 2), 0);
+    }
+
+    @Test(expectedExceptions = IllegalStateException.class, expectedExceptionsMessageRegExp = "type is not orderable")
```

Comment:
We avoid `@Test(expectedExceptions`, especially when test is not a one-liner.

Replace with `assertThatThrownBy`

Reply:
Got it. Replaced.